scientific_library/tvb/simulator/models/deco2020_transcriptional.py

```diff
@@ -0,0 +1,323 @@
+
+    J_i = NArray(
+        label=r":math:`J_{i}`",
+        default=numpy.array([1.0, ]),
+        domain=Range(lo=0.001, hi=2.0, step=0.001),
+        doc="""[nA] Local inhibitory current""")
+
+    W_i = NArray(
+        label=r":math:`W_i`",
+        default=numpy.array([0.7, ]),
+        domain=Range(lo=0.0, hi=1.0, step=0.01),
+        doc="""Inhibitory population external input scaling weight""")
+
+    I_o = NArray(
+        label=":math:`I_{o}`",
+        default=numpy.array([0.382, ]),
+        domain=Range(lo=0.0, hi=1.0, step=0.001),
+        doc="""[nA]. Effective external input""")
+
+    G = NArray(
+        label=":math:`G`",
+        default=numpy.array([2.0, ]),
+        domain=Range(lo=0.0, hi=10.0, step=0.01),
+        doc="""Global coupling scaling""")
+
+    lamda = NArray(
+        label=":math:`\lambda`",
+        default=numpy.array([0.0, ]),
+        domain=Range(lo=0.0, hi=1.0, step=0.01),
+        doc="""Inhibitory global coupling scaling""")
+
+    state_variable_range = Final(
+        default={
+            "S_e": numpy.array([0.0, 1.0]),
+            "S_i": numpy.array([0.0, 1.0]),
+            "H_e": numpy.array([0.0, 100]),
+            "I_e": numpy.array([0.0, 30.0])
+        },
+        label="State variable ranges [lo, hi]",
+        doc="Population firing rate")
+
+    # Used for phase-plane axis ranges and to bound random initial() conditions.
+    state_variable_boundaries = Final(
+        label="State Variable boundaries [lo, hi]",
+        default={"S_e": numpy.array([0.0, 1.0]),
+                 "S_i": numpy.array([0.0, 1.0]),
+                 "H_e": numpy.array([0.0, 150.0]),
+                 "I_e": numpy.array([0.0, 30.0])},
+        doc="""The values for each state-variable should be set to encompass
+            the boundaries of the dynamic range of that state-variable. Set None for one-sided boundaries""")
+
+    variables_of_interest = List(
+        of=str,
+        label="Variables watched by Monitors",
+        choices=('H_e', 'I_e', 'S_i', 'S_e'),
+        default=('H_e', 'I_e', 'S_i', 'S_e'),
+        doc="""default state variables to be monitored""")
+
+    state_variables = ['S_e', 'S_i', 'H_e', 'I_e']
+    _nvar = 4
+    cvar = numpy.array([0], dtype=numpy.int32)
+
+    def configure(self):
+        """  """
+        super(Deco2020Transcriptional, self).configure()
+        self.update_derived_parameters()
+
+    def initial(self, dt, history_shape, rng=numpy.random):
+        """Generates uniformly distributed initial conditions,
+        bounded by the state variable limits defined by the model.
+        """
+        ic = numpy.empty(history_shape)
+
+        ic[:, 0] = 0.001
+        ic[:, 1] = 0.001
+        ic[:, 2] = 0.001
+        ic[:, 3] = 0.001
+
+        return ic
+
+    def _numpy_dfun(self, state_variables, coupling, local_coupling=0.0):
+        r"""
+        """
+
+        S = state_variables[:, :]
+
+        S_e = S[:, 0]
```

Comment:
wrong indexing,  should be `S_e = S[0,:]

Reply:
Fixed